agent/api/task/task.go

```diff
@@ -379,6 +379,13 @@ func (task *Task) PostUnmarshalTask(cfg *config.Config,
 		return apierrors.NewResourceInitError(task.Arn, err)
 	}
 
+	if task.requiresFSxWindowsFileServerResource() {
+		if err := task.initializeFSxWindowsFileServerResource(cfg, credentialsManager, resourceFields); err != nil {
+			seelog.Errorf("Task [%s]: could not initialize FSx for Windows File Server resource: %v", task.Arn, err)
```

Comment:
nit: Having windows specific checks/logs in the platform agnostic file is confusing?

Reply:
the error message is not windows specific. also, i have both these checks returning false for linux in unsupported files. following same logic as checking for credspec resource.

Reply:
can this be `fsx` generic?
```
	if task.requiresFsxResource() {
		if err := task.initializeFsxResource(cfg, credentialsManager, resourceFields); err != nil {
```
I assume we'll need to do some of this for FSx Lustre as well, and this clears out Windows specifics in the agnostic file.

Reply:
I don't think that's feasible. FSx for Windows File Server and FSx Lustre would be different task resources and this method has initialize() per task resource.

Reply:
`fsxWindowsFileServer` is the product type -- it is technically "agnostic" ie it should apply to both Windows and Linux tasks.  
Please add a comment to this effect.

Reply:
done